### ISSUE TITLE
Shuffle training data in clientbase.py

### ISSUE THREAD

**Issue by matteo-pennisi** [TsingZ0/PFL-Non-IID#117]
Train dataset is not shuffled by default and this lead to problems when training models with batch norm
https://github.com/TsingZ0/PFL-Non-IID/blob/0a6e176947f403981933abaa834450eae8338518/system/flcore/clients/clientbase.py#L61C10-L61C10

```python
    def load_train_data(self, batch_size=None):
        if batch_size == None:
            batch_size = self.batch_size
        train_data = read_client_data(self.dataset, self.id, is_train=True)
        return DataLoader(train_data, batch_size, drop_last=True, shuffle=False) <--- 
```

**Comment by TsingZ0** [comment]
Thank you for pointing out this issue, I have fixed it. Please check it :-). 

Due to frequent version updates caused by the addition of new algorithms, it is possible for some old issues to reoccur.
